tools/evm/T8NTool/T8NTool.cs

```diff
@@ -0,0 +1,300 @@
+using Nethermind.Core.Specs;
+using Nethermind.Core.Test.Builders;
+using Nethermind.Crypto;
+using Nethermind.Db;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.JsonRpc;
+using Nethermind.Logging;
+using Nethermind.Serialization.Json;
+using Nethermind.Serialization.Rlp;
+using Nethermind.Specs;
+using Nethermind.Specs.Forks;
+using Nethermind.Specs.Test;
+using Nethermind.State;
+using Nethermind.State.Proofs;
+using Nethermind.Trie.Pruning;
+
+namespace Evm.T8NTool;
+
+public class T8NTool
+{
+    private readonly TxDecoder _txDecoder = new();
+    private readonly EthereumJsonSerializer _ethereumJsonSerializer = new();
+    private readonly LimboLogs _logManager = LimboLogs.Instance;
+
+    public int Execute(
+        string inputAlloc,
+        string inputEnv,
+        string inputTxs,
+        string outputAlloc,
+        string? outputBasedir,
+        string? outputBody,
+        string outputResult,
+        int stateChainId,
+        string stateFork,
+        string? stateReward,
+        bool traceMemory,
+        bool traceNoMemory,
+        bool traceNoReturnData,
+        bool traceNoStack,
+        bool traceReturnData)
+    {
+        try
+        {
+            T8NExecutionResult t8NExecutionResult = Execute(inputAlloc, inputEnv, inputTxs, stateFork, stateReward);
+
+            var stdoutObjects = new Dictionary<string, object>();
+            OutputObject(outputAlloc, outputBasedir, "alloc", t8NExecutionResult.Alloc, stdoutObjects);
+            OutputObject(outputResult, outputBasedir, "result", t8NExecutionResult.PostState, stdoutObjects);
+            OutputObject(outputBody, outputBasedir, "body", t8NExecutionResult.Body, stdoutObjects);
+
+            if (!stdoutObjects.IsNullOrEmpty())
+            {
+                Console.WriteLine(_ethereumJsonSerializer.Serialize(stdoutObjects, true));
+            }
+
+            return 0;
+        }
+        catch (IOException e)
+        {
+            throw new T8NException(e, ExitCodes.ErrorIO);
+        }
+        catch (JsonException e)
+        {
+            throw new T8NException(e, ExitCodes.ErrorJson);
+        }
+    }
+
+    private T8NExecutionResult Execute(
+        string inputAlloc,
+        string inputEnv,
+        string inputTxs,
+        string stateFork,
+        string? stateReward)
+    {
+        Dictionary<Address, AccountState> allocJson = _ethereumJsonSerializer.Deserialize<Dictionary<Address, AccountState>>(File.ReadAllText(inputAlloc));
+        EnvInfo envInfo = _ethereumJsonSerializer.Deserialize<EnvInfo>(File.ReadAllText(inputEnv));
+        Transaction[] transactions;
+        if (inputTxs.EndsWith(".json")) {
+            TransactionInfo[] txInfoList = _ethereumJsonSerializer.Deserialize<TransactionInfo[]>(File.ReadAllText(inputTxs));
+            transactions = txInfoList.Select(txInfo => txInfo.ConvertToTx()).ToArray();
+        } else {
```

Comment:
if not json we assume it is an RLP encoded tx List?

Reply:
What if a file not ending in either .rlp or .json was provided? We will consider it an rlp file?